README.md

```diff
@@ -112,6 +114,145 @@ plugins: [
+export default {
+	input: 'main.js',
+	plugins: [babel.generated()],
+	output: [
+		{ file: 'bundle.cjs.js', format: 'cjs' },
+		{ file: 'bundle.esm.js', format: 'esm' },
+	],
+};
+```
+
+If you only want to apply it to specific outputs, you can use it as an output plugin:
+
+```js
+// rollup.config.js
+import babel from 'rollup-plugin-babel';
+export default {
+	input: 'main.js',
+	output: [
+		{ file: 'bundle.js', format: 'esm' },
+		{
+			file: 'bundle.es5.js',
+			format: 'esm',
+			plugins: [babel.generated({ presets: ['@babel/env'] })],
+		},
+	],
+};
+```
+
+The `include`, `exclude` and `extensions` options are ignored when the when using `.generated()` will produce warnings, and there are a few more points to note that users should be aware of.
+
+You can also run the plugin twice on the code, once when processing the input files to transpile special syntax and once on the output to transpile to a lower compatibility target:
+
+```js
+// rollup.config.js
+import babel from 'rollup-plugin-babel';
+export default {
+	input: 'main.js',
+	plugins: [babel({ presets: ['@babel/preset-react'] })],
+	output: [
+		{
+			file: 'bundle.js',
+			format: 'esm',
+			plugins: [babel.generated({ presets: ['@babel/env'] })],
+		},
+	],
+};
+```
+
+### Finding Babel configuration options
+
+Besides Babel options that are passed directly to this plugin, `babel-generated(...)` will only respect project-wide [Babel configuration files](https://babeljs.io/docs/en/config-files) based on the current working directory when applying it to the generated code.
+
+### Using formats other than ES modules or CommonJS
+
+As `babel.generated(...)` will run _after_ Rollup has done all its transformations, it needs to make sure it preserves the semantics of Rollup's output format. This is especially important for Babel plugins that add, modify or remove imports or exports, but also for other transformations that add new variables as they can accidentally become global variables depending on the format. Therefore it is recommended that for formats other than `esm` or `cjs`, you set Rollup to use the `esm` output format and let Babel handle the transformation to another format, e.g. via
+
+```
+presets: [['@babel/env', { modules: 'umd' }], ...]
+```
+
+to create a UMD/IIFE compatible output. If you want to use `babel.generated(...)` with other formats, you need to specify `allowAllFormats: true` as plugin option:
+
+```js
+rollup.rollup({...})
+.then(bundle => bundle.generate({
+  format: 'iife',
+  plugins: [babel.generated({
+    allowAllFormats: true
+  })]
+}))
+```
+
+### Injected helpers
+
+By default, helpers e.g. when transpiling classes will be inserted at the top of each chunk. In contrast to when applying this plugin on the input files, helpers will not be deduplicated across chunks.
+
+You can however use external helpers by adding the `@babel/external-helpers` plugin, which will look for helpers on a global variable named `babelHelpers`:
```

Comment:
IMHO nowadays usage of `@babel/external-helpers` (thus global `babelHelpers` as mentioned here) is super rare and should be barely mentioned in the docs - or at least mentioned last. We shouldn't encourage its usage.

Reply:
Makes sense, I will reduce it to a side-note.